src/OrbitBase/include/OrbitBase/Typedef.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_TYPEDEF_H_
+#define ORBIT_BASE_TYPEDEF_H_
+
+#include <functional>
+#include <type_traits>
+#include <utility>
+
+namespace orbit_base {
+
+template <typename T, typename U>
+using EnableIfUConvertibleToT = std::enable_if_t<std::is_convertible_v<U, T>>;
+
+// Strong typedef
+template <typename Tag, typename T>
+class Typedef {
+ public:
+  const T* operator->() const { return &value_; }
+  T* operator->() { return &value_; }
+
+  const T& operator*() const& { return value_; }
+  T&& operator*() && { return std::move(value_); }
+  T& operator*() & { return value_; }
+
+  template <typename U, typename = EnableIfUConvertibleToT<T, U>>
+  explicit Typedef(U&& value) : value_(std::forward<U>(value)) {}
```

Comment:
Can you make `explicit` depend on whether `U` can be converted implicitly or explicitly to `T`?

This probably needs to overloads that are enabled/disable via SFINAE